File: tests/vb/Reference.Tests.VB.xUnit/SimpleIntegrationTest.vb
Imports System.Net.Http
Imports System.Threading.Tasks
Imports Xunit
' ReSharper disable InconsistentNaming

''' <summary>
''' Simple Integration Test
''' </summary>
Public Class SimpleIntegrationTest

    ''' <summary>
    ''' Simple Http Get OK
    ''' </summary>
    <Fact>
    Public Async Function HttpOK() As Task
        Using client = New HttpClient()
            Await client.GetAsync("http://www.google.com")
        End Using
    End Function

    ''' <summary>
    ''' Simple Http Get BadFormat
    ''' </summary>
    <Fact>
    Public Async Function HttpBadFormat() As Task
        Using client = New HttpClient()
            Await client.GetAsync("http://www.badUrl.c213")
        End Using
    End Function

    ''' <summary>
    ''' Simple Http Get KO
    ''' </summary>
    <Fact>
    Public Async Function HttpKO() As Task
        Using client = New HttpClient()
            Await client.GetAsync("http://localhost")
        End Using
    End Function

End Class

File: tests/csharp/Reference.Tests.CSharp.xUnit/SimpleIntegrationTest.cs
using System.Net.Http;
using System.Threading.Tasks;
using Xunit;
// ReSharper disable InconsistentNaming
// ReSharper disable xUnit1004

namespace Reference.Tests.CSharp.xUnit
{
    /// <summary>
    /// Simple Integration Test
    /// </summary>
    public class SimpleIntegrationTest
    {
        /// <summary>
        /// Simple Http Get OK
        /// </summary>
        [Fact]
        public async Task HttpOK()
        {
            using (var client = new HttpClient())
                await client.GetAsync("http://www.google.com");
        }

        /// <summary>
        /// Simple Http Get BadFormat
        /// </summary>
        [Fact]
        public async Task HttpBadFormat()
        {
            using (var client = new HttpClient())
                await client.GetAsync("http://www.badUrl.c213");
        }

        /// <summary>
        /// Simple Http Get KO
        /// </summary>
        [Fact]
        public async Task HttpKO()
        {
            using (var client = new HttpClient())
                await client.GetAsync("http://localhost");
        }
    }
}